packages/transaction-controller/src/utils/transaction-type.ts

```diff
@@ -88,28 +88,28 @@ export async function determineTransactionType(
 }
 
 /**
- * Attempts to decode transaction data using ABIs for three different token standards: ERC20, ERC721, ERC1155.
+ * Parses transaction data using ABIs for three different token standards: ERC20, ERC721, ERC1155 and USDC.
  * The data will decode correctly if the transaction is an interaction with a contract that matches one of these
  * contract standards
  *
  * @param data - Encoded transaction data.
  * @returns A representation of an ethereum contract call.
  */
-function getMethodName(data?: string): string | undefined {
-  if (!data || data.length < 10) {
-    return undefined;
-  }
-
-  const fourByte = data.substring(0, 10).toLowerCase();
-
-  for (const interfaceInstance of [
+export function decodeTransactionData(data: string) {
+  const interfaces = [
     ERC20Interface,
     ERC721Interface,
     ERC1155Interface,
     USDCInterface,
-  ]) {
+  ];
+
+  if (!data) {
+    return undefined;
+  }
+
+  for (const iface of interfaces) {
     try {
-      return interfaceInstance.getFunction(fourByte).name;
+      return iface.parseTransaction({ data });
```

Comment:
This will break incoming transactions as we only have the first four bytes available, hence why we used `getFunction` instead of `parseTransaction`.

Could we use have an optional boolean in this function that `getMethodName` uses to use `getFunction` instead and returns just an object with a `name` property?